Add v2 application deployment docs (#11746)
hmcts/cnp-flux-config: docs/app-deployment-v2.md

# Application Configuration

> >Note: Depending on the [Migration Status](/README.md#Migration status) and the environment you are adding config, you should also do [v1 setup](app-deployment.md)

Reviewer: link isn't working for migration status

The section covers how to configure a new application in this repository to deploy to various environments. We use [kustomize](https://github.com/kubernetes-sigs/kustomize) for templating/patching manifests in this repo. 

**It is important to follow below discussed naming convention, file and folder names should match application helm release name**

> Note: Some of these scripts need you to [install yq](https://mikefarah.gitbook.io/yq/)

Reviewer: english isn't right on above, this could work of changing it to a 'Prerequisites'
```suggestion
Note: You need to [install yq](https://mikefarah.gitbook.io/yq/) for these scripts
```

kind: HelmRelease
metadata:
  name: <component-name>
spec:
  values:
    java:
      image: hmctspublic.azurecr.io/<product>/<component>:<latest pr-123-tag>   #{"$imagepolicy": "flux-system:<env>-<component-name>"}
      #Example below
      #image: hmctspublic.azurecr.io/draft-store/service:pr-123-20210807222025   #{"$imagepolicy": "flux-system:demo-draft-store-service"}
```

- Create a image policy like below with your pr number (pr-332 taken as example) along side your HelmRelease file.

```yaml
apiVersion: image.toolkit.fluxcd.io/v1alpha2
kind: ImagePolicy
metadata:
  name: <env>-<component-name>
  annotations:
    hmcts.github.com/prod-automated: disabled
spec:
  filterTags:
    pattern: '^pr-332-[a-f0-9]+-(?P<ts>[0-9]+)'
    extract: '$ts'
  policy:
    alphabetical:
      order: asc
  imageRepositoryRef:
    name: <component-name>
```
- Add `- ../<component-name>/<env>-image-policy.yaml` to `resources` in automation kustomization `apps/<your-namespace>/automation/kustomization.yaml`

### Deploying an app to single cluster

- It is recommended to add all applications to both clusters as one of the cluster can be removed from operation at any point. 

> - If you still have a use case, it has to be dealt exceptionally with help of [#rpe](https://hmcts-reform.slack.com/archives/C8SR5CAMU)

Reviewer: platops

## Testing in Local

> If you want to find the effective yaml that will get applied to an environment for your namespace, you can [install kustomize](https://kubernetes-sigs.github.io/kustomize/installation/)  and run build commands.

Reviewer: ```suggestion
If you want to find the effective yaml that will get applied to an environment for your namespace, you can [install kustomize](https://kubernetes-sigs.github.io/kustomize/installation/) and run build commands.
```
